src/state/DB.js

```diff
@@ -39,9 +50,65 @@ class DB {
         });
     }
 
-    updateFile() {
+    verifyPassword(password) {
+        return hash.verify(password, this.config.hashed_password);
+    }
+
+    isPasswordNotSet() {
+        return this.config.hashed_password === null;
+    }
+
+    setPassword(password) {
+        if (password === null) {
+            this.setConfig({...this.config, ...{"hashed_password": null}});
+        } else {
+            this.setConfig({...this.config, ...{"hashed_password": hash.generate(password)}});
+        }
+
+        this.updateConfigFile();
+    }
+
+    setConfig(config) {
```

Comment:
seems like you always call `updateConfigFile` right after `setConfig` anyways, so maybe you should just put `updateConfigFile` in `setConfig`?

Reply:
yeah, I was trying to match the logic to the users file, but that makes more sense